engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb

```diff
@@ -1,20 +1,27 @@
 # frozen_string_literal: true
 
 module OrderManagement
-  module Subscriptions
+  module Order
     class StripeScaPaymentAuthorize
+      include FullUrlHelper
+
       def initialize(order)
         @order = order
         @payment = OrderPaymentFinder.new(@order).last_pending_payment
       end
 
-      def call!
+      def call!(redirect_url = full_order_path(@order), send_emails = true)
         return unless @payment&.checkout?
 
-        @payment.authorize!
+        @payment.authorize!(redirect_url)
 
         @order.errors.add(:base, I18n.t('authorization_failure')) unless @payment.pending?
 
+        if send_emails && @payment.cvv_response_message.present?
+          PaymentMailer.authorize_payment(@payment).deliver_now
+          PaymentMailer.authorization_required(@payment).deliver_now
+        end
```

Comment:
:warning: If I checked correctly, this class is used from `StripeRedirect` and that one isn't called from a background job :warning:

Reply:
This is true; however, in `StripeRedirect` (ie in the checkout) the `send_emails` flag is set to false.... in any case, though, I'm going to refactor that flag param away.